level2/restaurants/kubo_restaurants.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+# Restaurant class
+class Restaurant
+  attr_reader :name, :items
+
+  def initialize(name)
+    @name = name
+    @items = []
+  end
+
+  def add_items(*items)
+    @items += items
+  end
+
+  def remove_item(item_name)
+    @items.reject! { |item| item.name == item_name }
+  end
+
+  def place_order(order_items, discount: 0)
+    order_price = 0
+    order_items.each do |order_item|
+      return DOUBLE_DISCOUNT_ERROR if order_item[:discount] && discount.positive?
+
+      checked_item = @items.find { |i| i.name == order_item[:name] }
+      return UNAVAILABLE_ITEM_ERROR unless checked_item&.in_stock?(order_item[:count])
+
+      checked_item.stock -= order_item[:count]
```

Comment:
this has one more problem, when you order 2 items, and the first is in stock while the second is not, the first items stock will go down eventhough the order doesnt go through.

Reply:
Ok, I fixed it so the stock change happens only after the validation of all items